keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/NUTSTest.java

```diff
@@ -25,6 +28,46 @@ public void setup() {
         random = new KeanuRandom(1);
     }
 
+    @Category(Slow.class)
+    @Test
+    public void canRecordStatisticsFromSamples() {
+        double mu = 0.0;
+        double sigma = 1.0;
+        double initStepSize = 1;
+        int maxTreeHeight = 4;
+        BayesianNetwork simpleGaussian = MCMCTestDistributions.createSimpleGaussian(mu, sigma, 3, random);
+
+        NUTS nuts = NUTS.builder()
+            .adaptEnabled(false)
+            .initialStepSize(initStepSize)
+            .random(random)
+            .maxTreeHeight(maxTreeHeight)
+            .saveStatistics(true)
+            .build();
+
+        nuts.getPosteriorSamples(
+            simpleGaussian,
+            simpleGaussian.getLatentVertices(),
+            2
+        );
+
+        Statistics statistics = nuts.getStatistics();
+
+        List<Double> stepSize = statistics.get("stepSize");
+        List<Double> logProb = statistics.get("logProb");
+        List<Double> meanTreeAccept = statistics.get("meanTreeAccept");
+        List<Double> treeSize = statistics.get("treeSize");
+
+        Assert.assertTrue(stepSize.get(0) == initStepSize);
+        Assert.assertTrue(stepSize.get(1) == initStepSize);
+        Assert.assertTrue(logProb.get(0) < 0);
+        Assert.assertTrue(logProb.get(1) < 0);
+        Assert.assertTrue(meanTreeAccept.get(0) >= 0 && meanTreeAccept.get(0) <= 1);
+        Assert.assertTrue(meanTreeAccept.get(1) >= 0 && meanTreeAccept.get(1) <= 1);
```

Comment:
assertThat(meanTreeAccept, everyItem(both(greaterThanOrEqualTo(0.)).and(lessThanOrEqualTo(1.))));